include/gridtools/stencil_composition/sid/sid_helpers.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include "../../meta.hpp"
+#include "concept.hpp"
+#include "delegate.hpp"
+#include "multi_shift.hpp"
+
+namespace gridtools {
+    namespace sid {
+        namespace sid_helpers_impl_ {
+            template <class Sid>
+            class shifted_sid : public delegate<Sid> {
+                sid::ptr_holder_type<Sid> m_origin;
+
+                friend GT_META_CALL(sid::ptr_holder_type, Sid) sid_get_origin(shifted_sid &obj) { return obj.m_origin; }
+
+              public:
+                template <class Offsets>
+                constexpr shifted_sid(Sid &&original_sid, Offsets &&offsets) noexcept
```

Comment:
remove `constexpr` here. it already uses `multi_shift` that can not be `constexpr`.